core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -120,6 +121,28 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
     }
   }
 
+  // Estimate app frequency based off of all applications in this run
+  private def estimateAppFrequency(
+    appsSum: Seq[QualificationSummaryInfo]): Seq[QualificationSummaryInfo] = {
+    var appFrequency = Map[String, Double]()
```

Comment:
should be a val, you aren't recreating the map

Reply:
Map values are being reassigned in
`appFrequency += (sum.appName -> (1.0 + appFrequency.getOrElse(sum.appName, 0.0)))
`